app/models/organisation.rb

```diff
@@ -81,9 +81,21 @@ def self.search_by_keyword(keyword)
      self.where(contains_description(keyword).or(contains_name(keyword)))
   end
 
-  def self.filter_by_category(category_id)
-    return all unless category_id.present?
-    self.joins(:categories).where(is_in_category(category_id)) #do we need to sanitize category_id?
+  def self.filter_by_categories(category_ids)
```

Comment:
@mtc2013 The ability to switch so easily between `and` and `or` is amazing. This inspired me to do a refactor. [Here's a diff of the specific commits](https://github.com/johnnymo87/LocalSupport/compare/fa6257c76dc27d92f19db28d9994963dbc96d836...aae15d3bb27772f8b9f84b6e6c526fb25ebd0387). Thoughts?